src/store/accounts/nonce.ts

```diff
@@ -31,10 +32,18 @@ export const currentAccountNonceState = atom(get => {
   const confirmedTransactions = get(currentAccountConfirmedTransactionsState);
   const pendingTransactions = get(currentAccountMempoolTransactionsState);
   const lastLocalNonce = get(currentAccountLocalNonceState);
+  const latestLocallySumbmittedNonce = get(currentAccountLocallySubmittedLatestNonceState);
 
   const apiNonce = get(lastApiNonceState);
-  // We try to use the api nonce first since it will be the most accurate value
-  if (typeof apiNonce === 'number') return apiNonce;
+  if (typeof apiNonce === 'number') {
+    if (latestLocallySumbmittedNonce) {
+      // if we have a locally submitted nonce, and it's higher than the api nonce, use that
+      if (latestLocallySumbmittedNonce.toNumber() > apiNonce)
+        return latestLocallySumbmittedNonce.toNumber();
```

Comment:
@aulneau If this is the last locally submitted nonce, shouldn't this be latestLocallySubmittedNonce + 1 ?